core/includes/install.core.inc

```diff
@@ -1409,28 +1383,52 @@ function install_select_language(&$install_state) {
  *
  * @ingroup forms
  */
-function install_select_language_form($form, &$form_state, $files) {
+function install_select_language_form($form, &$form_state, $files, $install_state) {
   include_once BACKDROP_ROOT . '/core/includes/standard.inc';
   include_once BACKDROP_ROOT . '/core/includes/locale.inc';
 
   $standard_languages = standard_language_list();
   $select_options = array();
   $languages = array();
 
-  foreach ($files as $file) {
-    if (isset($standard_languages[$file->langcode])) {
-      // Build a list of select list options based on files we found.
-      $select_options[$file->langcode] = $standard_languages[$file->langcode][1];
-    }
-    else {
-      // If the language was not found in standard.inc, display its langcode.
-      $select_options[$file->langcode] = $file->langcode;
+  if (count($install_state['translations']) > 1) {
+    $files = $install_state['translations'];
+  }
+  else {
+    $files = [];
+  }
+  $select_options = [];
+  $browser_options = [];
+
+  $form['#title'] = 'Choose language';
+
+  // Build a select list with language names in native language for the user
+  // to choose from. And build a list of available languages for the browser
+  // to select the language default from.
+  // Select lists based on all standard languages.
+  foreach ($standard_languages as $langcode => $language_names) {
```

Comment:
Don't we already have a function for that?
IIRC there's country_get_list().

Reply:
We're looking at languages here, not countries.

Reply:
> We're looking at languages here, not countries.

Doh... :see_no_evil: 

There's standard_language_list(). :wink: Did you already have a look at it's function comment? Looks interesting for our use-case.

Reply:
Yep that is the function we're using to build the language list in all cases.